File: android/app/src/main/kotlin/com/example/player/model/Song.kt
package com.example.player.model

import java.io.Serializable
import java.util.HashMap

class Song(
        val id: Long,
        val title: String,
        val artist: String = "",
        val album: String = "",
        val albumArt: String = "",
        val track: Int = 0,
        val duration: Long = 0,
        val data: String
): Serializable {

    var elapsed: Int = 0

    fun toMap(): HashMap<String, Any> {
        val map = HashMap<String, Any>()
        map["id"] = id as Any
        map["title"] = title as Any
        map["artist"] = artist as Any
        map["album"] = album as Any
        map["albumArt"] = albumArt as Any
        map["track"] = track as Any
        map["duration"] = duration as Any
        map["data"] = data as Any
        map["elapsed"] = elapsed as Any
        return map
    }

    companion object {
        fun fromMap(map: Map<String, Any>?): Song {
            return Song(
                    id = map?.get("id").toString().toLong(),
                    title = map?.get("title").toString(),
                    artist = map?.get("artist").toString(),
                    album = map?.get("album").toString(),
                    albumArt = map?.get("albumArt").toString(),
                    track = map?.get("track").toString().toInt(),
                    duration = map?.get("duration").toString().toLong(),
                    data = map?.get("data").toString()
            )
        }
    }
}

File: lib/model/song.dart

class Song {
  int id;
  String title;
  String artist;
  String album;
  String albumArt;
  int track;
  int duration;
  String data;
  int elapsed;

  Song(this.id, this.title, this.artist, this.album, this.albumArt,
      this.track, this.duration, this.data);

  Song.fromMap(Map m) {
    id = m["id"];
    title = m["title"];
    artist = m["artist"];
    album = m["album"];
    albumArt = m["albumArt"];
    track = m["track"];
    duration = m["duration"];
    data = m["data"];
    elapsed = m["elapsed"];
  }

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      "id": id,
      "title": title,
      "artist": artist,
      "album": album,
      "albumArt": albumArt,
      "track": track,
      "duration": duration,
      "data": data,
    };
  }

  String durationInSeconds() {
    var text = Duration(milliseconds: duration).toString().split('.').first;
    if (text.startsWith('0:'))
      text = text.substring(2, text.length);
    return text;
  }

  String elapsedInSeconds() {
    var text = Duration(milliseconds: elapsed).toString().split('.').first;
    if (text.startsWith('0:'))
      text = text.substring(2, text.length);
    return text;
  }

  int elapsedAsPosition() {
    if (elapsed > duration)
      return duration;
    return elapsed;
  }
}